### DISCUSSION TITLE
[Proposal]: Allow default value for out parameter

### DISCUSSION THREAD

**Discussion by heku** [dotnet/csharplang#3989] [Language Ideas] [answered]
# Allow default value for out parameter

* [x] Proposed
* [ ] Prototype: Not Started
* [ ] Implementation: Not Started
* [ ] Specification: Not Started

## Summary
Allow default value for `out` parameter

## Detailed design
Allow add default value for `out` parameter so that we can change the code below
```c#
bool Method(out string value) {
  if(xxxx){
    value = "xxx";
    return true;
  }
  value = null;
  return false;
}
```
to
```c#
bool Method(out string value = null) {
  if(xxxx){
    value = "xxx";
    return true;
  }
  return false;
}
```

This feature seems only need C# compiler level update by compiling parameter `out string value = null` to `out string value` with one more line at the first of the method body `value = null`

**Comment by YairHalberstadt** [comment]
What's the advantage of:

```csharp
bool Method(out string value = null) {
  if(xxxx){
    value = "xxx";
    return true;
  }
  return false;
}
```

over

```csharp
bool Method(out string value) {
  value = null;
  if(xxxx){
    value = "xxx";
    return true;
  }
  return false;
}
```

**Comment by 333fred** [comment]
I'm going to convert this to a discussion. We generally want proposals to be more complete. Adding more around the motivation here would help sell the idea, as I agree with Yair that I don't see much of a benefit.

**Comment by miloush** [comment, marked as answer]
Duplicate of https://github.com/dotnet/roslyn/issues/1757 resp. https://github.com/dotnet/csharplang/issues/1033

**Reply by YairHalberstadt** [reply]
Note that https://github.com/dotnet/csharplang/issues/1033 is championed

**Reply by YairHalberstadt** [reply]
And was rejected
